### PR TITLE
Fix weak refs

### PR DESCRIPTION
**Pull request by AngellusMortis**
Fixes https://github.com/AngellusMortis/pyunifiprotect/issues/267

### CODE REVIEW

File: pyunifiprotect/data/bootstrap.py [hunk 1]
```diff
@@ -488,7 +490,7 @@ def _handle_ws_error(self, action: Dict[str, Any], err: Exception) -> None:
             try:
                 model_type = ModelType(action["modelKey"])
                 device_id = action["id"]
-                asyncio.create_task(self.refresh_device(model_type, device_id))
+                self._refresh_tasks.append(asyncio.create_task(self.refresh_device(model_type, device_id)))
```
Comment (bdraco) [inline pyunifiprotect/data/bootstrap.py:493]: We could make refresh tasks a set and use add_callback_done on the task to discard the task from the set when it's done. The task is passed as the argument to add_callback_done so the callback can be self._refresh_tasks.discard

Comment (AngellusMortis) [reply pyunifiprotect/data/bootstrap.py:493]: If multiple refreshes happen at once, that could cause a race condition. A task could be queued later, but completed first.

Comment (bdraco) [reply pyunifiprotect/data/bootstrap.py:493]: I'm probably not explaining that well on my mobile. I'll give you a better example when I'm at by desk

In the mean time. Here is how bleak solved the issue https://github.com/hbldh/bleak/commit/e01ae590df381097df12e9353064627960db4b09

Comment (AngellusMortis) [reply pyunifiprotect/data/bootstrap.py:493]: Oh, the task is passed back into done callback. Which that makes sense. 

The python docs actually as an example of this: https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task

**Review by bdraco — APPROVED** [review]
tested 👍

**Pull request merged**